src/metabase/pulse.clj

```diff
@@ -399,4 +400,7 @@
                   ;; to fetch the Pulse.
                   pulse/hydrate-notification
                   (merge (when channel-ids {:channel-ids channel-ids})))]
+    ;; validate all email domains in the pulse channels before sending, in case this is a "Send email now" kind of test
+    (doseq [channel (:channels pulse)]
+      (pulse-channel/validate-email-domains channel))
```

Comment:
we only want to validate new Dashboard Subscriptions/Alerts going forward, existing ones that used to work should get grandfathered in and continue to work